iup/__init__.py

```diff
@@ -996,112 +1010,42 @@ def predict(
 
 
 #### prediction output ####
-class Forecast(ValidateData):
+class QuantileForecast(ValidatedData):
     """
-    Abstract class for all the forecast data type.
+    Class for forecast with quantiles.
+    Save for future.
     """
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-    # Can add validate to enforce the specific schema of Forecast when more Forecast types are added
+    # Must be named as "quantileXX" except the date column
```

Comment:
As noted above, columns for grouping factors seem necessary but don't fit this naming system.

Reply:
Just added two assert functions that only check on numerical columns, then adding grouping factors won't cause issues. See 8ead6e304c8f18304e32fbaa24b7f5b216ea247c